action/protocol/staking/protocol.go

```diff
@@ -278,6 +282,19 @@ func (p *Protocol) CreatePreStates(ctx context.Context, sm protocol.StateManager
 		if err := p.voteReviser.Revise(csm, blkCtx.BlockHeight); err != nil {
 			return err
 		}
+		if p.voteReviser.NeedCorrectCands(blkCtx.BlockHeight) {
```

Comment:
current logic: only if https://github.com/iotexproject/iotex-core/pull/3688/files#diff-6bac7263ac5bdd5a72fb2bbb66bff76ebd0b794b0f998ed62bfafab7bba319e6R277 is true, the function will be called. therefore, `NeedRevise` should return true for all `correct candidates heights"

Reply:
fixed